packages/loader/container-loader/src/collabWindowTracker.ts

```diff
@@ -69,9 +69,14 @@ export class CollabWindowTracker {
         }
 
         this.opsCountSinceNoop++;
-        if (this.opsCountSinceNoop >= this.NoopCountFrequency) {
-            this.submitNoop(false /* immediate */);
-            return;
+        if (this.opsCountSinceNoop === this.NoopCountFrequency) {
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            Promise.resolve().then(() => {
+                this.submitNoop(false /* immediate */);
```

Comment:
Can you please add assert here that this.opsCountSinceNoop >= this.NoopCountFrequency ?
I believe it should be true statement and having assert like that will help capture regressions in the future if (let's say) we schedule many promises like that in a row (not just one).